code/modules/mob/living/carbon/human/species/unathi.dm

```diff
@@ -128,3 +125,34 @@
 		"appendix" = /obj/item/organ/internal/appendix,
 		"eyes" =     /obj/item/organ/internal/eyes/unathi
 		)
+
+/datum/species/unathi/ashwalker/on_species_gain(mob/living/carbon/human/H)
+	var/datum/action/innate/ignite/ash_walker/fire = new()
+	fire.Grant(H)
+
+/datum/species/unathi/ashwalker/on_species_loss(mob/living/carbon/human/H)
+	..()
+	for(var/datum/action/innate/ignite/ash_walker/fire in H.actions)
+		fire.Remove(H)
+
+/datum/action/innate/ignite/ash_walker
+	desc = "You form fire in your mouth, allowing you to.. light cigarettes."
+
+/datum/action/innate/ignite/ash_walker/Activate()
```

Comment:
there's a lot of copypasted code between this and its parent. instead would you add a `cooldown_duration` var and a `welding_fuel_used` var or on `/datum/action/innate/ignite` rather than hardcoding these values in.